packages/application-components/src/components/dialogs/internals/dialog-header.tsx

```diff
@@ -21,13 +22,28 @@ export const TextTitle = (props: TTextTitleProps) => (
   </Text.Headline>
 );
 
+const HiddenEmptyDialogTitle = () => (
+  <div aria-hidden={true} style={{ display: 'none' }}>
+    <DialogTitle />
+  </div>
+);
+
 type TitleProps = Pick<Props, 'title'>;
 const Title = (props: TitleProps) => {
-  if (typeof props.title === 'string') {
-    return <TextTitle title={props.title} />;
-  } else {
-    return <>{props.title}</>;
-  }
+  return (
+    <>
+      {typeof props.title === 'string' ? (
+        <TextTitle title={props.title} />
+      ) : (
+        props.title
+      )}
+      {/* FIXME: Temporary workaround for https://github.com/radix-ui/primitives/issues/2986
+        Radix UI's DialogContent requires rendering a DialogTitle, which renders as <h2>.
```

Comment:
What does it happen if we don't provide a `DialogTitle`? Does the component throw an error? a warning? is it only in development mode?

Reply:
Thanks, 
We're getting a console error in the browser and tests in CI fail with
<img width="621" alt="image" src="https://github.com/user-attachments/assets/718d78b2-30fb-43be-b7f9-12480f14977f" />

> is it only in development mode?

Yes

Reply:
What I was thinking is that, since this won't happen in production, we would be injecting an element in the DOM that we don't need and we're also adding this extra complexity to the code. Perhaps we can just [globally silence](https://github.com/commercetools/merchant-center-application-kit/blob/main/packages/jest-preset-mc-app/setup-test-framework.js#L87) that error in our tests as we know where it's coming from and that it's safe.

Reply:
Sure, we could suppress it here, that's a valid option.
But it would still fail in the mc-fe CI, right? So we'd need to silence it in multiple places.

Reply:
I pushed this change https://github.com/commercetools/merchant-center-application-kit/pull/3710/commits/789577efc86a5348fbfe0339a4ff41966d8f7224 (not 100% if this is what you meant), but it seems to me it's causing [fails in the CI downstream](https://app.circleci.com/pipelines/github/commercetools/merchant-center-frontend/99515/workflows/215b34f8-e4e1-49a5-bb48-686fff24ddde/jobs/2352459).
Please mind that we'll need to add additional setup for the app using vitest (it uses our dialogs too).

Reply:
Ok. I was just trying to know if we could avoid that extra HTML element but seems it's not that easy so I think it's better to keep it like you have it before the last commit.
(_Perhaps is something to look as a follow-up_)